src/hooks/api/useBuildGetter.ts

```diff
@@ -0,0 +1,20 @@
+import { useCallback } from 'react'
+import useFetch from 'hooks/useFetch'
+import { GetHookResponse } from './types'
+
+export const useBuildGetter = <T, U extends Array<unknown> = Array<any>>(
+  paramBuilder: (...args: U) => [string, Record<string, any>]
```

Comment:
Isn't it better if the second argument were optional?

Reply:
this does result in some strange syntax, let me know what you think

Reply:
nevermind, managed to solve it in a cleaner way!